activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb

```diff
@@ -11,8 +11,14 @@ def mysql2_connection(config)
 
       config[:username] = 'root' if config[:username].nil?
       config[:flags] ||= 0
+
       if Mysql2::Client.const_defined? :FOUND_ROWS
-        config[:flags] |= Mysql2::Client::FOUND_ROWS
+        case config[:flags]
+        when Integer
+          config[:flags] |= Mysql2::Client::FOUND_ROWS
+        when Array
+          config[:flags].push "FOUND_ROWS".freeze
+        end
```

Comment:
Would move the `Integer` case to an `else` fallback. Now if non-integer flags are passed, this silently no-ops rather than raising on missing `#|` method.